server/ssl.c

```diff
@@ -79,6 +90,77 @@ AP_DECLARE(int) ap_ssl_conn_is_ssl(conn_rec *c)
     return r;
 }
 
+static int ssl_engine_set(conn_rec *c,
+                          ap_conf_vector_t *per_dir_config,
+                          int proxy, int enable)
+{
+    if (proxy) {
+        return ap_ssl_bind_outgoing(c, per_dir_config, enable) == OK;
+    }
+    else if (module_ssl_engine_set) {
+        return module_ssl_engine_set(c, per_dir_config, 0, enable);
+    }
+    else if (enable && module_ssl_proxy_enable) {
```

Comment:
Do we need this given that we are not in a proxy case here (per `if (proxy)` early return above)?

Reply:
Not sure. Theoretically, someone could have called with such parameters, although not from our source tree.

Reply:
I think yes as the old ssl_engine_set in mod_ssl.c also does things in case that `proxy` is 0 e.g. switching `sslconn->disabled` based on `enable`:

```C
static int ssl_engine_set(conn_rec *c,
                          ap_conf_vector_t *per_dir_config,
                          int proxy, int enable)
{
    SSLConnRec *sslconn;
    int status;

    if (proxy) {
        sslconn = ssl_init_connection_ctx(c, per_dir_config, 1);
    }
    else {
        sslconn = myConnConfig(c);
    }

    status = ssl_engine_status(c, sslconn);

    if (proxy && status == DECLINED) {
        if (enable) {
            SSLSrvConfigRec *sc = mySrvConfig(sslconn->server);
            ap_log_cerror(APLOG_MARK, APLOG_ERR, 0, c, APLOGNO(01961)
                          "SSL Proxy requested for %s but not enabled "
                          "[Hint: SSLProxyEngine]", sc->vhost_id);
        }
        sslconn->disabled = 1;
    }
    else if (sslconn) {
        sslconn->disabled = !enable;
    }

    return status != DECLINED;
}
```

Reply:
Ah ok right, the old ssl_proxy_enable() called ssl_engine_set() actually.